microprofile/testing/junit5/src/main/java/io/helidon/microprofile/testing/junit5/ConfigBlob.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.testing.junit5;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UncheckedIOException;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Objects;
+import java.util.Properties;
+
+import io.helidon.config.mp.MpConfigSources;
+import io.helidon.config.yaml.mp.YamlMpConfigSource;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * Defines the configuration as a String in {@link #content()} for the
+ * given type.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Inherited
+public @interface ConfigBlob {
```

Comment:
I'm not convinced with that name as "blob" usually relates to binary stuff.

Can we come-up with something better ?
```java
@ConfigBlock
@ConfigChunk
@ConfigFragment
```

---

Also, should we use the prefix `Add` to be consistent with the current `AddConfig` annotation ?

```java
@AddConfig(key = "foo", value = "bar")
@ConfigBlock("""
    foo=bar
    bar=foo
""")
```

Compared to:

```java
@AddConfig(key = "foo", value = "bar")
@AddConfigBlock("""
    foo=bar
    bar=foo
""")
```

Compared to (re-using the `@AddConfig` annotation):

```java
@AddConfig(key = "foo", value = "bar")
@AddConfig(block = """
    foo=bar
    bar=foo
""")
```

Reply:
`@AddConfigBlock` is probably better, because with the IDE you can autocomplete, and it will help users to know what do you have available just writting `@Add`

Regarding re-using `@AddConfig` I don't really support this. There are some combinations that makes no sense, for example when you specify `key`, `value` and `type`. It could confuse the user.

Reply:
You are right that re-using the existing annotation could be confusing.

However with annotations, users have to be aware of them, they cannot be found naturally by navigating the code.
One could argue that having a new annotation isn't necessarily better for end users.

I don't care that much though :) As long as we pick a good name.